src/components/auth/signup/signup.controller.js

```diff
@@ -1,12 +1,42 @@
 export default class SignUpController {
   /** @ngInject */
-  constructor($auth, $state, $toast) {
+  constructor($auth, $state, $toast, $fetch) {
     Object.assign(this, {
-      $auth, $state, $toast,
+      $auth, $state, $toast, $fetch, credentials: {},
     });
   }
 
   /**
+   * Check whether the email is used.
+   * @return void
+   */
+  checkEmail() {
+    if (this.form.email.$valid) {
+      this.checkEmailRequest();
+    }
+  }
+
+  /**
+   * Send a request to server for check the email.
+   * @return void
+   */
+  checkEmailRequest() {
```

Comment:
For re-usable, it is better to put this method in signup module instead of in the controller.